src/msg/async/rdma/RDMAConnectedSocketImpl.cc

```diff
@@ -233,18 +239,19 @@ void RDMAConnectedSocketImpl::handle_connection() {
         ldout(cct, 10) << __func__ << " server is already active." << dendl;
         return ;
       }
+      r = activate();
+      assert(!r);
       r = infiniband->send_msg(cct, tcp_fd, my_msg);
       if (r < 0) {
         ldout(cct, 1) << __func__ << " server ack failed." << dendl;
         dispatcher->perf_logger->inc(l_msgr_rdma_handshake_errors);
         fault();
         return ;
       }
-      r = activate();
-      assert(!r);
     } else { // ack from client
       connected = 1;
-      cleanup();
+      ldout(cct, 10) << __func__ << " handshake of rdma is done. server connected: " << connected << dendl;
+      //cleanup();
```

Comment:
Server will not recv msg from client with tcp_fd after the process of cleanup(). so the server's tcp_fd will be in the CLOSE_WAIT state when the client disconnect the connection actively. There are so many fds in CLOSE_WAIT state that no extra fds to link up when the ceph is running after a period of time. 
I remove the cleanup() to make server can still recieve msg from client to aviod the CLOSE_WAIT.
To avoid the leak of socket after I remove this cleanup(), I add the process that "if(1 == connected){...}" in handle_connection().
So, server will process the disconnected msg with handle_connection(), and it will trigger the "if(1 == connected){...}" operation at this point when the when the client disconnects the TCP link actively. After the branch of "if(1 == connected){...}" in handle_connection(), server will close its socket and disconnect the connection.